packages/transaction-controller/src/TransactionController.ts

```diff
@@ -556,14 +621,117 @@ export class TransactionController extends BaseControllerV1<
 
     this.addPendingTransactionTrackerListeners();
 
+    // when transactionsController state changes
+    // check for pending transactions and start polling if there are any
+    this.subscribe(this.#checkForPendingTransactionAndStartPolling);
+
+    // TODO once v2 is merged make sure this only runs when
+    // selectedNetworkClientId changes
     onNetworkStateChange(() => {
       log('Detected network change', this.getChainId());
+      this.pendingTransactionTracker.onStateChange();
       this.onBootCleanup();
     });
 
     this.onBootCleanup();
+    this.messagingSystem.subscribe(
+      'NetworkController:stateChange',
```

Comment:
Can the above `onNetworkStateChange` logic go in this listener since it would remove an additional constructor callback?

Reply:
`onNetworkStateChange` doesnt call `listener` with any arguments like the `patch`, maybe we can go the other way, remove `onNetworkStateChange` entirely.

I think the thought here was that `onNetworkStateChange` should really be from `networkDidChange` event not stateChange.

Which is a bug we've fixed in a couple places, `provider` is not set before stateChange.

Reply:
Understood, I misread that this was a network state change but it's a general state update.